src/ensembl/src/shared/components/help-popup/useHelpArticle.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect } from 'react';
+
+import config from 'config';
+
+import useApiService from 'src/shared/hooks/useApiService';
+
+import {
+  HelpArticle,
+  HelpVideo,
+  RelatedArticle,
+  SlugReference,
+  PathReference
+} from './types';
+
+const getQuery = (params: { reference: SlugReference | PathReference }) => {
+  if ('slug' in params.reference) {
+    return `slug=${params.reference.slug}`;
+  } else {
+    return `path=${encodeURIComponent(params.reference.path)}`;
+  }
+};
+
+type Reference = SlugReference | PathReference;
+
+export type ArticleReference = {
+  type: 'article';
+  reference: Reference;
+};
+
+export type VideoReference = {
+  type: 'video';
+  youtube_id: string;
+};
+
+export type RelatedItems = {
+  articles: RelatedArticle[];
+  videos: HelpVideo[];
+};
+
+export type CurrentArticle = HelpArticle & { type: 'article' };
+export type CurrentVideo = HelpVideo & { type: 'video' };
+export type CurrentItem = CurrentArticle | CurrentVideo;
+
+export const emptyRelatedItems: RelatedItems = {
+  articles: [],
+  videos: []
+};
+
+const useHelpArticle = (reference: ArticleReference | VideoReference) => {
+  const [currentHelpItem, setCurrentHelpItem] = useState<CurrentItem | null>(
+    null
+  );
+  const [relatedItems, setRelatedItems] = useState<RelatedItems | null>(null);
+
+  const query = reference.type === 'article' ? getQuery(reference) : null;
+  const { helpApiHost } = config;
+  const url = query ? `${helpApiHost}/api/article?${query}` : '';
+
+  const { data: article, loadingState } = useApiService<HelpArticle>({
+    endpoint: url,
+    skip: !url
+  });
+
+  useEffect(() => {
+    if (
+      article &&
+      reference.type === 'article' &&
+      (('slug' in reference.reference &&
+        reference.reference.slug === article.slug) ||
+        ('path' in reference.reference &&
+          reference.reference.path === article.path))
```

Comment:
Why do we need to check if the article `slug` or `path` matches with the passed in reference?

Reply:
Just an extra check to make sure that we are putting the correct article in the state, and not, by any chance an outdated article from a previous render.